lib/connect/api/base.js

```diff
@@ -5,69 +5,210 @@
  */
 const _ = require('lodash');
 const querystring = require('querystring');
+const { Query } = require('rql/query');
 const { HttpError, APIError } = require('./errors');
 
+
 /**
- * The BaseService class is a base class from which inherits every endpoint service
+ * The GenericResource class is a base class from which inherits every endpoint service
  * of the Cloud Blue Connect API.
  * It provides utility functions usefull in subclasses.
  */
-class BaseService {
+class GenericResource {
   /**
    *
-   * @param {AbstractHttpAdapter} adapter A concrete implementation of http adapter.
+   * @param {ConnectClient} client   A ConnectClient instance.
+   * @param {string}        baseUri  The base URI of the resource
    */
-  constructor(adapter) {
-    this._adapter = adapter;
+  constructor(client, baseUri) {
+    this._client = client;
+    this._baseUri = baseUri;
```

Comment:
In fact there is a really cool thing that you might prefer comparing to all this work with strings
It's URL object - https://nodejs.org/api/url.html might be really useful

Reply:
I agree that it's a best option to use the url module to manipulate urls.